doc/cprover-manual/index.md

```diff
@@ -32,4 +32,8 @@
    * [Integration into Build Systems with goto-cc](goto-cc/)
    * [Integration with Visual Studio builds](visual-studio/)
 
-10. [The CPROVER API Reference](api/)
+10. Solvers
+
+    * [Incremental SMT solver](smt2-incr/)
```

Comment:
Leaving it up to @kroening to make a call here, for he maintains the rendered version of this documentation. My own view is that we should not discuss a particular solver as part of this manual.

Reply:
I consider the "official" documentation now to be automatically rendered here: https://diffblue.github.io/cbmc/ Is that not the case? Will that not be the case for this file?

Where would you propose we discuss things specific to a particular solver?

Reply:
I agree that this one (github.io) is the right place; it will _also_ include the CPROVER manual (which is otherwise rendered on cprover.org), but isn't limited to that manual.

Reply:
> I consider the "official" documentation now to be automatically rendered here: https://diffblue.github.io/cbmc/ Is that not the case? Will that not be the case for this file?

I can confirm that if I checkout and build the documentation locally as will be done on the documentation publishing job, then this new documentation will be included. Therefore it should be uploaded to the github.io page by github actions after this PR is merged.

The links to follow to find it will be -

1. User Guide in the side bar.
2. The link to "A short manual" in that pages's content.
3. The new page link will be under item 10.

> Where would you propose we discuss things specific to a particular solver?

I think this is the right place to add this documentation. Most information about the SAT solvers for instance wouldn't belong here, because they are compile/link time swappable not run-time swappable. However I think this list started by Enrico could be the right place to add documentation of the the dimacs file and external sat solver support, when someone takes the time to write it.